inception-imls-stringmatch/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/stringmatch/StringMatchingRecommender.java

```diff
@@ -74,13 +72,18 @@ public StringMatchingRecommender(Recommender aRecommender,
         traits = aTraits;
     }
 
-    public void pretrain(List<GazeteerEntry> aData)
+    public void pretrain(List<GazeteerEntry> aData, RecommenderContext aContext)
     {
+    	Trie<DictEntry> dict = createTrie();
+    	
         if (aData != null) {
-            pretrainData = aData;
+            for (GazeteerEntry entry : aData) {
+                learn(dict, entry.text, entry.label);
+            }
         }
-        else {
-            pretrainData = emptyList();
+        
+        if (aContext != null) {
```

Comment:
Why can the context be null?

Reply:
Right, should not. It's not tested anywhere else, so removed it here too.